### PR TITLE
Add monotonicity check for some functions

### PR DESCRIPTION
**Pull request by junli1026**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

After #3009 is checked in,  functions as follows need to have monotonic checking:
1. arithmetic div
2. arithmetic mul
3. date number functions
4. week date funcction
5. floor
6. ceil
7. sign

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by junli1026** [comment]
@zhyass  could you please help to take a look ?

File: common/functions/src/scalars/dates/number_function.rs [hunk 1]
```diff
@@ -458,6 +561,10 @@ where
         }?;
         Ok(number_array)
     }
+
+    fn get_monotonicity(&self, args: &[Monotonicity]) -> Result<Monotonicity> {
```
Comment (zhyass) [inline common/functions/src/scalars/dates/number_function.rs:565]: Maybe
```
if T::MAYBE_MONOTONIC {
    return Ok(Monotonicity::default());
}
Ok(Monotonicity::clone_without_range(&args[0]))
```
is better.

File: common/functions/src/scalars/arithmetics/arithmetic_mul.rs [hunk 2]
```diff
@@ -33,8 +34,154 @@ impl ArithmeticMulFunction {
...
 
-    pub fn get_monotonicity(_args: &[Monotonicity]) -> Result<Monotonicity> {
-        //TODO
-        Ok(Monotonicity::default())
+    pub fn get_monotonicity(args: &[Monotonicity]) -> Result<Monotonicity> {
+        arithmetic_mul_div_monotonicity(args, DataValueArithmeticOperator::Mul)
+    }
+}
+
+pub fn arithmetic_mul_div_monotonicity(
```
Comment (zhyass) [inline common/functions/src/scalars/arithmetics/arithmetic_mul.rs:42]: Need to consider the case where the divisor is 0.

File: common/functions/src/scalars/arithmetics/arithmetic_mul.rs [hunk 3]
```diff
@@ -33,8 +34,154 @@ impl ArithmeticMulFunction {
...
+    let f_x = &args[0];
+    let g_x = &args[1];
+
+    match (f_x.is_constant, g_x.is_constant) {
+        // both f(x) and g(x) are constant
+        (true, true) => Ok(Monotonicity::create_constant()),
+
+        //f(x) is constant
+        (true, false) => {
+            if f_x.gt_eq_zero()? {
```
Comment (zhyass) [inline common/functions/src/scalars/arithmetics/arithmetic_mul.rs:82]: 12/g(x), also need check the domain of g(x), when g(x)>0 decreasing, when g(x)<0 increasing.

File: common/functions/src/scalars/arithmetics/arithmetic_mul.rs [hunk 4]
```diff
@@ -33,8 +34,154 @@ impl ArithmeticMulFunction {
...
+            } else {
+                // unknown monotonicity
+                Ok(Monotonicity::default())
+            }
+        }
+
+        // neither f(x) nor g(x) are constant, like abs(x) *|- (x+12)
+        (false, false) => {
+            if f_x.is_monotonic && g_x.is_monotonic {
+                let f_x_gt_eq_zero = f_x.gt_eq_zero()?;
```
Comment (zhyass) [inline common/functions/src/scalars/arithmetics/arithmetic_mul.rs:127]: Maybe we can create a function `fn max(left, right)-> Result<i8>`, 1: left>right, 0: left=right, -1 left<right

**Comment by BohuTANG** [comment]
@mergify update

File: common/functions/src/scalars/maths/sign.rs [hunk 5]
```diff
@@ -101,6 +103,31 @@ impl Function for SignFunction {
...
+    fn get_monotonicity(&self, args: &[Monotonicity]) -> Result<Monotonicity> {
+        let mono = args[0].clone();
+        if mono.is_constant {
+            return Ok(Monotonicity::create_constant());
+        }
+
+        // check whether the left/right boundary is numeric or not.
+        let is_boundary_numeric = |boundary: Option<DataColumnWithField>| -> bool {
+            if let Some(column_field) = boundary {
+                column_field.field().data_type().is_numeric()
```
Comment (zhyass) [inline common/functions/src/scalars/maths/sign.rs:116]: column_field.data_type().is_numeric()

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
@mergifyio update

**Pull request merged**